.github/workflows/approved-for-ci-run.yml

```diff
@@ -69,16 +69,43 @@ jobs:
         with:
           ref: main
           token: ${{ secrets.CI_ACCESS_TOKEN }}
+      
+      - name: Look for existant PR
+        id: "get_pr"
+        run: |
+          ALREADY_CREATED="$(gh pr --repo ${GITHUB_REPOSITORY} list --head ${BRANCH} --base main --json number --jq '.[].number')"
+          echo "ALREADY_CREATED=$ALREADY_CREATED" >> $GITHUB_OUTPUT
+      
+      - name: Get changed labels
+        id: "get_labels"
+        env:
+          ALREADY_CREATED: ${{ steps.get_pr.outputs.ALREADY_CREATED }}
+        run: |
+          LABELS_TO_ADD=
+          LABELS_TO_REMOVE=
+          if [ -n "$ALREADY_CREATED" ]; then
+            LABELS_TO_REMOVE=$(comm -23 <(gh pr --repo ${GITHUB_REPOSITORY} view $ALREADY_CREATED --json labels --jq '.labels.[].name'| grep -E '^run' | sort) <(gh pr --repo ${GITHUB_REPOSITORY} view $PR_NUMBER --json labels --jq '.labels.[].name' |  grep -E '^run' | sort ))
+            LABELS_TO_REMOVE=${LABELS_TO_REMOVE//run-e2e-tests-in-draft/}
+            LABELS_TO_REMOVE=$(echo $LABELS_TO_REMOVE | awk 'BEGIN {list=""} {for (i=1;i<=NF;i++) if (length($i)>0) if (length(list)>0) list=list "," $i; else list=$i;} END {print list}')
+            LABELS_TO_ADD=$(comm -13 <(gh pr --repo ${GITHUB_REPOSITORY} view $ALREADY_CREATED --json labels --jq '.labels.[].name'|  grep -E '^run' |sort) <(gh pr --repo ${GITHUB_REPOSITORY} view $PR_NUMBER --json labels --jq '.labels.[].name' |  grep -E '^run' | sort ))
+            LABELS_TO_ADD=$(echo $LABELS_TO_ADD | awk 'BEGIN {list=""} {for (i=1;i<=NF;i++) if (length($i)>0) if (length(list)>0) list=list "," $i; else list=$i;} END {print list}')
+          fi
+          echo "LABELS_TO_ADD=$LABELS_TO_ADD" >> $GITHUB_OUTPUT
+          echo "LABELS_TO_REMOVE=$LABELS_TO_REMOVE" >> $GITHUB_OUTPUT
 
       - run: gh pr checkout "${PR_NUMBER}"
 
       - run: git checkout -b "${BRANCH}"
 
       - run: git push --force origin "${BRANCH}"
+        if: steps.get_pr.outputs.ALREADY_CREATED == ''
 
       - name: Create a Pull Request for CI run (if required)
-        env:
+        id: "create_pr"
+        env: 
           GH_TOKEN: ${{ secrets.CI_ACCESS_TOKEN }}
+          ALREADY_CREATED: ${{ steps.get_pr.outputs.ALREADY_CREATED }}
```

Comment:
If we don't use it in the step itself, we don't need to create an env var for it

Reply:
done